src/clojure_ttt/computer.clj

```diff
@@ -1,12 +1,36 @@
 (ns clojure-ttt.computer
   (use [clojure-ttt.board]))
 
-  (defn opposite-player [player]
-    (if (= player "x") "o" "x"))
-
-  (defn minimax [board player]
-      (some 
-        #(cond
-          (any-wins? (mark-board board % (opposite-player player))) %
-          (any-wins? (mark-board board % player)) %)
-          (available-positions board)))
+  (defn- score [board depth marks]
+    (cond (and (any-wins? board) (= (last-move board) (first marks))) depth
+    (and (any-wins? board) (= (last-move board) (second marks))) (- depth)
+    :else 0))
+
+  (defn- all-moves-played [board mark positions]
+    (map #(mark-board board % mark) positions))
+
+  (defn- initial-score [marks player]
+    (if (= (first marks) player) {:score -1000 :board nil} {:score 1000 :board nil}))
+
+  (defn- favours-computer? [marks player new-score current-best]
+    (cond 
+    (= (first marks) player) (> new-score (:score current-best)) 
+    :else (< new-score (:score current-best))))
+
+  (defn- minimax [board player depth marks]
+    (if (or (= depth 0) (or (any-wins? board) (draw? board)))
```

Comment:
this could be made more readable by extracting into a method: e.g. (if (or (full-board? depth) (game-over? board)))